pyiceberg/table/update/__init__.py

```diff
@@ -727,6 +727,14 @@ class AssertRefSnapshotId(ValidatableTableRequirement):
     ref: str = Field(...)
     snapshot_id: Optional[int] = Field(default=None, alias="snapshot-id")
 
+    @model_serializer
+    def ser_model(self) -> dict[str, Any]:
+        return {
+            "type": self.type,
+            "ref": self.ref,
+            "snapshot-id": self.snapshot_id,
+        }
```

Comment:
nit: is there a way to call super() or the default serializer? we just want to explicitly override for `snapshot-id`. 

otherwise, we'd have to remember to change to this function everytime we add a new variable